nemo/collections/nlp/models/text_classification_model.py

```diff
@@ -0,0 +1,232 @@
+# Copyright 2020 NVIDIA. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import Dict, Optional
+
+import torch
+from torch.utils.data import DataLoader
+
+from nemo.collections.common.losses import CrossEntropyLoss
+from nemo.collections.common.tokenizers.bert_tokenizer import NemoBertTokenizer
+from nemo.collections.nlp.data.text_classification import TextClassificationDataDesc, TextClassificationDataset
+from nemo.collections.nlp.modules.common import SequenceClassifier
+from nemo.collections.nlp.modules.common.common_utils import get_pretrained_lm_model
+from nemo.core.classes import typecheck
+from nemo.core.classes.modelPT import ModelPT
+from nemo.core.neural_types import NeuralType
+
+__all__ = ['TextClassificationModel']
+
+
+class TextClassificationModel(ModelPT):
+    @property
+    def input_types(self) -> Optional[Dict[str, NeuralType]]:
+        return self.bert_model.input_types
+
+    @property
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        return self.classifier.output_types
+
+    def __init__(
+        self,
+        data_dir: str,
+        pretrained_model_name: str,
+        bert_config: str,
+        num_output_layers: int,
+        fc_dropout: float,
+        class_balancing: bool,
+    ):
+        """
+        Initializes the BERTTextClassifier model.
+        Args:
+            data_dir (str): the path to the folder containing the data
+            pretrained_model_name (str): name of the BERT model to be used as the encoder
+            bert_config (str): The path to the config file for the BERT encoder, it should be None to use the default configs
+            num_output (int)_layers (int): number of the linear layers of the mlp head on the top of the encoder
+            fc_dropout (float): the dropout used for the mlp head
+            class_balancing (bool): enables the weighted class balancing of the loss, may be used for handling unbalanced classes
+        """
+
+        # init superclass
+        super().__init__()
+
+        data_desc = TextClassificationDataDesc(data_dir=data_dir, modes=["train", "test", "dev"])
+
```

Comment:
Can we control modes through args?

Reply:
This data descriptor reads all the data to create the stats. That was the default and it just ignores the one does not exists, but I would make it to just process the train and the eval file.

Reply:
I just remember there were issues with some datasets that don't have all 3 sets, I remember we added smth to control this, just want to make sure it's fixed. How does the user select eval set? is it passed via args?

Reply:
Fixed that one by giving a warning that that file does not exist. It is passed via the args but not passed into the init of the model as it is currently for model params. If we move data param into the model init, we can be more selective which one exactly to convert. One small difference is that if select the modes here, then if test has a label which does not exist in the train and eval, it would give error when tried on test. Currently we process all the files to capture all the labels.